hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileConfig.java

```diff
@@ -34,19 +40,23 @@
   private boolean dropBehindCacheCompaction;
   private Configuration hadoopConf;
   private BloomFilter bloomFilter;
-
-  // This is private in CacheConfig so have been copied here.
-  private static boolean DROP_BEHIND_CACHE_COMPACTION_DEFAULT = true;
+  private KeyValue.KVComparator hfileComparator;
 
   public HoodieHFileConfig(Configuration hadoopConf, Compression.Algorithm compressionAlgorithm, int blockSize,
                            long maxFileSize, BloomFilter bloomFilter) {
     this(hadoopConf, compressionAlgorithm, blockSize, maxFileSize, CacheConfig.DEFAULT_PREFETCH_ON_OPEN,
-        HColumnDescriptor.DEFAULT_CACHE_DATA_IN_L1, DROP_BEHIND_CACHE_COMPACTION_DEFAULT, bloomFilter);
+        HColumnDescriptor.DEFAULT_CACHE_DATA_IN_L1, DROP_BEHIND_CACHE_COMPACTION_DEFAULT, bloomFilter, defaultHFileComparator);
+  }
+
+  public HoodieHFileConfig(Configuration hadoopConf, Compression.Algorithm compressionAlgorithm, int blockSize,
```

Comment:
have 1 constructor that always takes a comparator?

Reply:
Yeah we can do that if we set default value in the storage config.

Reply:
done.